### PR TITLE
refactor(anvil): use ethers native Filter types

### PR DESCRIPTION
**Pull request by mattsse**
Blocked by https://github.com/gakonst/ethers-rs/pull/1389

## Motivation
uses ethers-rs Filter type directly

### CODE REVIEW

**Comment by gakonst** [comment]
good for ethers bump @mattsse

**Review by gakonst — APPROVED** [review]
minor nits - feel free to merge otherwise. I feel like it's worth doing another pass at the anvil types and see which ones we can replace/upstream with ethers-rs types. 

iirc `NameOrAddress` was a problem but I don't recall the exact problem

File: anvil/src/eth/backend/mem/mod.rs [hunk 1]
```diff
@@ -629,7 +628,7 @@ impl Backend {
         }
 
         if let Some(fork) = self.get_fork() {
-            let filter = filter.into();
+            let filter = filter;
```
[the remark this answers is not in the document]
Comment (mattsse) [reply anvil/src/eth/backend/mem/mod.rs:631]: right, no conversion necessary anymore

File: anvil/core/src/eth/subscription.rs [hunk 2]
```diff
@@ -36,7 +36,7 @@ pub enum SubscriptionParams {
     /// no `params`
     None,
     /// `Filter` parameters.
-    Logs(Filter),
+    Logs(Box<Filter>),
```
[the remark this answers is not in the document]
Comment (mattsse) [reply anvil/core/src/eth/subscription.rs:39]: yeh clippy large size diff, since Filter holds an array.

good nit, rewriting it as new type

**Comment by mattsse** [comment]
makes sense to give it another pass

> iirc `NameOrAddress` was a problem but I don't recall the exact problem

main issue was that we'd need some `match + unreachable!()` to get the address since we only getting addresses.

but perhaps fine since it only allows derserializing `Address`

**Comment by gakonst** [comment]
> but perhaps fine since it only allows derserializing Address

yep

**Pull request merged**